contracts/core/EntryPoint.sol

```diff
@@ -141,6 +149,24 @@ contract EntryPoint is IEntryPoint, StakeManager {
         _compensate(beneficiary, collected);
     }
 
+    function simulateExecution(UserOperation calldata op) external override {
+
+        UserOpInfo memory opInfo;
+
+        (uint256 deadline, uint256 paymasterDeadline,) = _validatePrepayment(0, op, opInfo, SIMULATE_FIND_AGGREGATOR);
```

Comment:
You do pass the `SIMULATE_FIND_AGGREGATOR` into the `_validatePrepayment`, so the method is, so to speak, "simulation-aware".
However, for the purpose of `deadline == SIG_VALIDATION_FAILED` check, it is not aware it is being run in a simulation and returns a failure. I think this is inconsistent.
Maybe if you did not have to reset the `deadline` and `paymasterDeadline` in the `simulateExecution` you would not have to separate the `_validatePrepayment` and `_validateDeadline`? This separation seems to me a bit arbitrary anyway.

Reply:
yes, the method is has limited knowledge of simulation, but I wanted to keep that to a minimum, since it is executed on-chain for each transaction.
small deadline values revert (on expiration) . so using "1" value will always revert.  that's the rationale of using a "deadline" value to mark a (different) failure (signature)
if you look at validateUserOp, it only calls validatePrepayment, NOT validateDeadline (since it returns the deadlines, for the bundler to check off-chain)